cmd/helm3.go

```diff
@@ -120,7 +120,26 @@ func (d *diffCmd) template(isUpgrade bool) ([]byte, error) {
 		flags = append(flags, "--set-string", stringValue)
 	}
 	for _, valueFile := range d.valueFiles {
-		flags = append(flags, "--values", valueFile)
+		if strings.TrimSpace(valueFile) == "-" {
+			bytes, err := ioutil.ReadAll(os.Stdin)
+			if err != nil {
+				return nil, err
+			}
+
+			tmpfile, err := ioutil.TempFile("", "helm-diff-stdin-values")
+			if err != nil {
+				return nil, err
+			}
+			defer os.Remove(tmpfile.Name())
+
+			if _, err := tmpfile.Write(bytes); err != nil {
+				return nil, err
+			}
```

Comment:
Did this change actually work for you?

I'm afraid we're missing `tmpfile.Close()` here, that I think is required to ensure writes are persisted. We should call `Close()` regardless of err is nil or not so wrapping it into another function call would be easier to read.

See `writeExistingValues` for your reference!

Reply:
Yes, it worked pretty well. I got your point regarding closing the tempfile. tbh i don't want to overcomplicate it by adding one more function, so just added one more if condition.

Reply:
LGTM. Thanks for your effort!